### DISCUSSION TITLE
Quality of life: Recharging oxygen tanks

### DISCUSSION THREAD

**Discussion by DeoNoN** [FakeFishGames/Barotrauma#7971] [Ideas and suggestions]
Just a few simple tweaks: **1)** being able to change oxygen tanks of diving suits without needing to take them out of the lockers. **2)** a key to allow them to be taken in hands directly (I.E. LControl + Pick) , without you having to wear it first, exit the locker menu, and then double click it to put it on your hands.

1) When replacing the oxygen tanks of suits, sometimes is a bit tedious having to take the suit out of the locker, to then change the tanks, and putting it in again. This can be a bit confusing for newcomers, as you can not put the suit in your hands while using the locker. Also a bit annoying, as this takes more time than necessary, not for being a complex or hard task, just because of its inconvenience. I suggest this could be easily fixed by having a box that shows the oxygen tank equiped when hovering the mouse over the suit, as it is shown in the -poorly drown- image A, and as it works in your inventory, you could easily change the tanks just by dragging it in and out. I think this won't harm the gameplay and make it less clunky.

![IMG A](https://user-images.githubusercontent.com/95893172/147850896-b97cfda2-6e95-4c51-941b-bc0906d9cb9e.png)

2) As it is adressed in the first "problem", you do not have an option to take the suit in your hands directly, you have to wear it, then exit the menu, to wear it off by double clicking it, because if you do it while using the locker, you will put it in again, this path is represented in the -again poorly drawn- B image as the color RED. This could be an intended feature, but I don't think that is the case, because you already have the option to manipulate multiple inventories, for example managing a crate, a cabinet and a tool belt, all at the same time. A simple solution could be using a combination of keys like "LControl + Double click" or "LShift + Double click", as suits can not be stacked, so that will not override the function of picking one and half items respectively, this path is represented in with the color BLUE int he B image. Another fix can be showing the "extra slot" of hands in your inventory HUD, so you can drag the suit to it.

![IMG 2](https://user-images.githubusercontent.com/95893172/147851279-41c64aab-909e-4985-a3bf-1cf7e28c7b7e.png)

Final thoughts: I am aware these are minor issues, however I think is a good desing philosophy making the controls more convenient and clear for an already complex and rich game. Moreover, I belive these fixes could be easily implemented and not troubeling, but I am not into coding, so I might be missjudging the matter. Thanks for your time, and sorry if my explanations weren't as clear as I would like to (also, sorry for the horrible paint edit), cheers.

**Comment by MaaiG** [comment]
It's true! You can replace a suit’s oxigen tank with a simple drag-and-drop but you can’t take a tank out of a suit until you put on a suit!

P.S. Associated https://github.com/Regalis11/Barotrauma/discussions/7444

**Comment by MaaiG** [comment]
1. Need add the interactive oxygen tank icon to the suit GUI. You can drag and drop it to the inventory!
**Suggestion:**
![Image 001_22-48](https://user-images.githubusercontent.com/93348499/147974316-f4b79f9a-4550-4fdf-8a49-ae99dd831846.png)

**Comment by leonard84** [comment]
re 1) As oxygen tanks of suits in diving cabinets are recharged automatically there is little need to manually change the tank IMHO.

re 2) I agree that a being able to hit "e" directly on the cabinet without first having to open it then drag the suite out of the cabinet to the floor to then be able to pick it up with "e" would be really nice.
